src/editorDecorator.ts

```diff
@@ -0,0 +1,139 @@
+
+
+import * as vscode from "vscode";
+import {
+    DecorationRenderOptions,
+    TextEditorDecorationType,
+  } from "vscode";
+
+import {
+    testNodeType,
+} from "./testData";
+
+import {
+    getRangeOption,
+  } from "./utilities";
+
+const path = require("path");
+  
+
+// This is used in the package.json to control the display of context menu items
+// Search for 'vectorcastTestExplorer.testableLineList' in package.json to see where we reference it
+export var testableLineList: number[] = [];
+
+var testableFunctionDecorationType: TextEditorDecorationType;
+var testableFunctionOptions: DecorationRenderOptions;
+var testableFunctionsDecorations:vscode.DecorationOptions[] = [];
+
+export function initializeTestDecorator(context:vscode.ExtensionContext) {
+    
+    testableFunctionOptions = {
+        light: {gutterIconPath: context.asAbsolutePath("./images/light/beaker-plus.svg")},
+        dark: {gutterIconPath: context.asAbsolutePath("./images/dark/beaker-plus.svg")}
+    };
+    testableFunctionDecorationType = vscode.window.createTextEditorDecorationType(testableFunctionOptions);
+}
+
+
+interface unitDataType {
+    enviroPath:string,
+    enviroName:string,
+    unitName:string,
+    lineMap:Map<number, string>,
+}
+
+let unitAndFunctionMap: Map<string, unitDataType> = new Map();
+
+export function updateFunctionDataForFile(
+    enviroPath:string,
+    fileName: string,
+    functionList: string[]) {
+
+    // functionList is a list of json items with fields for "name" and "startLine"
+    let lineMap:Map<number, string> = new Map();
+    for (let i = 0; i < functionList.length; i++) {
+        const functionInfo:any = functionList[i];
```

Comment:
Useless type annotation

Reply:
You've marked a variable as type `:any`, which just means "this is any type" (so you might as well not have it)

Reply:
Wow, fair enough.